master/internal/api_trials.go

```diff
@@ -531,6 +536,80 @@ func (a *apiServer) PostTrialProfilerMetricsBatch(
 	return &apiv1.PostTrialProfilerMetricsBatchResponse{}, errs.ErrorOrNil()
 }
 
+func (a *apiServer) TrialPreemptionSignal(
+	req *apiv1.TrialPreemptionSignalRequest,
+	resp apiv1.Determined_TrialPreemptionSignalServer,
+) error {
+	trial, err := a.trialActorFromID(int(req.TrialId))
+	if err != nil {
+		return err
+	}
+
+	id := uuid.New()
```

Comment:
it's because once i type the trial's ref as `chan<- bool` and the listener's ref as `<-chan bool`, afaik there is no way to recover the `chan<- bool` from the `<-chan bool` (those types only convert in one direction, and using the recv chan in the send chan map is a compile error)

Reply:
i could probably use reflect to do it 😄

Reply:
ah, I didn't know this.  uuid is fine.